backend/app/emails.py

```diff
@@ -0,0 +1,135 @@
+from email.mime.text import MIMEText
+from email.utils import formataddr
+from enum import Enum
+from smtplib import SMTP_SSL as SMTP
+from typing import Any
+
+from github import Github
+from gitlab import Gitlab
+from jinja2 import Environment, PackageLoader, select_autoescape
+from pydantic import BaseModel
+from sqlalchemy import select
+
+from . import models
+from .config import settings
+from .db import get_json_key
+
+template_env = Environment(
+    loader=PackageLoader("app", "email_templates"),
+    autoescape=select_autoescape(default=True),
+)
+
+
+class EmailCategory(str, Enum):
+    SECURITY_LOGIN = "security_login"
+    MODERATION_HELD = "moderation_held"
+    MODERATION_APPROVED = "moderation_approved"
+    MODERATION_REJECTED = "moderation_rejected"
+
+
+class EmailInfo(BaseModel):
+    user_id: int | None
+    app_id: str | None
+    category: EmailCategory
+    subject: str
+    template_data: dict[str, Any]
+
+
+def _get_gitlab_email(account, gitlab_url) -> str | None:
+    from .logins import refresh_oauth_token
+
+    access_token = refresh_oauth_token(account)
+    with Gitlab(gitlab_url, oauth_token=access_token) as gl:
+        gl.auth()
+        return gl.user.public_email
+
+
+def _get_email_address(user: models.FlathubUser, db) -> str | None:
+    if gh_user := models.GithubAccount.by_user(db, user):
+        gh = Github(gh_user.token)
+        return gh.get_user().email
+
+    if gl_user := models.GitlabAccount.by_user(db, user):
+        return _get_gitlab_email(gl_user, "https://gitlab.com")
+
+    if gl_user := models.GnomeAccount.by_user(db, user):
+        return _get_gitlab_email(gl_user, "https://gitlab.gnome.org")
+
+    if gl_user := models.KdeAccount.by_user(db, user):
+        return _get_gitlab_email(gl_user, "https://invent.kde.org")
+
+
+def _create_message(
+    user: models.FlathubUser, info: EmailInfo, db
+) -> tuple[str, MIMEText]:
+    email = _get_email_address(user, db)
+    full_subject = info.subject
```

Comment:
Having written an email system at work this week too, I would like to see us prefix the subject, if the mail is coming from a test system. (which we don't have right now)

I tihkn you should be able to use the `ENV` env var for that?

Reply:
That's a good idea. We don't currently set ENV in the docker-compose.yml though, I suppose we should.

Reply:
Keep in mind, that the docker compose is only for development and isn't used on the server. It should still get the env.

Reply:
This has since been added